### PR TITLE
Restore timeout to keep compatible with formater CI runs, based on community feedback

### PR DESCRIPTION
**Pull request by TomasVotruba**
Based on various feedback after 0.19 release, it seems the best way to tackle timeout issues is to restore original 120 seconds.

Related to:

* https://github.com/rectorphp/rector/issues/8403#issuecomment-1892418947
* https://github.com/rectorphp/rector/issues/8404
* https://github.com/rectorphp/rector/issues/8396

### CODE REVIEW

File: src/Config/RectorConfig.php [hunk 1]
```diff
@@ -88,7 +88,7 @@ public function enableCollectors(): void
      * Defaults in sync with https://phpstan.org/config-reference#parallel-processing
      * as we run PHPStan as well
      */
-    public function parallel(int $processTimeout = 60, int $maxNumberOfProcess = 32, int $jobSize = 20): void
+    public function parallel(int $processTimeout = 120, int $maxNumberOfProcess = 32, int $jobSize = 20): void
```
Comment (samsonasik) [inline src/Config/RectorConfig.php:91]: max number process need to be 16, jobsize to be 15

Comment (TomasVotruba) [reply src/Config/RectorConfig.php:91]: Let's try one thing at a time at the moment, to see how it lands

**Comment by staabm** [comment]
Thanks!

**Comment by TomasVotruba** [comment]
Thanks for being patient with live testing :pray: 

We're now looking for ways to make Rector less memory greeding. I have a feeling it takes to much more memory than PHPStan, despite being quite similar. 

We could optimized file loading (spotted by @samsonasik ), maybe delegate more type resolving to PHPStan, as we used quite deffensive approach when we had docblock rules, and more. Creative ways and experiments with results as welcomed :rocket:

**Pull request merged**